go/vt/vtgate/tx_conn.go

```diff
@@ -153,8 +153,19 @@ func (txc *TxConn) commit2PC(ctx context.Context, session *SafeSession) error {
 		return err
 	}
 
+	var qs queryservice.QueryService
+	if UsingLegacyGateway() {
```

Comment:
We really need to remove the legacy gateway, or add more capabilities to it. these checks are really not good to spread out like this

Reply:
agreed :(